uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/controllers/request.controller.spec.js

```diff
@@ -703,6 +710,35 @@
           it('does not contain admin in the list of managees', function () {
             expect(_.find(controller.managedContacts, { 'id': adminId })).toBeUndefined();
           });
+
+          describe('on absence period change', function () {
+            beforeEach(function () {
+              $rootScope.$broadcast('LeaveRequestPopup::absencePeriodChanged');
+            });
+
+            it('starts reloading entitlements', function () {
+              expect(controller.loading.entitlements).toBeTruthy();
+            });
+
+            describe('once it started reloading entitlements', function () {
+              beforeEach(function () {
+                spyOn($rootScope, '$emit').and.callThrough();
+                $rootScope.$digest();
+              });
+
+              it('loads entitlements', function () {
```

Comment:
it's important also to test that the entitlements specifically for *the new period* had been requested

Reply:
Done.